Update project level docs (#1734)
paritytech/parity-bridges-common: docs/polkadot-kusama-bridge-overview.md

# Polkadot <> Kusama Bridge Overview

This document describes how we use all components, described in the [High-Level Bridge Documentation](./high-level-overview.md),
to build the XCM bridge between Kusama and Polkadot. In this case, our components merely work as a XCM transport
(like XCMP/UMP/HRMP), between chains that are not a part of the same consensus system.

The overall architecture may be seen in [this diagram](./polkadot-kusama-bridge.html).

## Bridge Hubs

> All operations at relay chain are expensive. Ideally all non-mandatory trasnsactions must happen on parachains.

Reviewer: ```suggestion
All operations at relay chain are expensive. Ideally all non-mandatory transactions must happen on parachains.
```

If we won't take any fees, it would mean that the sender is not obliged to pay for its messages. And Bridge Hub
collators (and, maybe, "treasury") are not receiving any payment for including transactions. More about this later,
in the [Who is Rewarding Relayers](#who-is-rewarding-relayers) section.

### Message Delivery Confirmation Rewards

In addition to the "zero-cost" message delivery transactions, the relayer is also rewarded for:

- delivering every message. The reward is registered during delivery confirmation transaction at the Source Bridge
  Hub.;

- submitting delivery confirmation transaction. The relayer may submit delivery confirmation that e.g. confirms
  delivery of four messages, of which the only one (or zero) messages is actually delivered by this relayer. It
  receives some fee for confirming messages, delivered by other relayers.

Both rewards may be claimed using the `pallet_bridge_relayers::claim_rewards` call at the Source Bridge Hub.

### Who is Rewarding Relayers

Obviously, there should be someone who is paying relayer rewards. We want bridge transactions to have a cost, so we

> can't use fees for rewards. Instead, the parachains, that are using the bridge, have a mutual agreement, that they
> are paying each other costs.

Reviewer: With the sovereign accounts model, I don't think there is the mutual agreement anymore. Like you say below, Statemine will pay _all_ relayer rewards for messages coming from Statemine (it pays on each side from its sov account on that respective BH). Statemint doesn't have to pay anything to receive messages from Statemine, right?

At most, Statemint might offer free XCM execution for messages with Computed Origin == Statemine, but in terms of relayer rewards it is doing nothing, right?

How about:
```suggestion
can't use fees for rewards. Instead, the parachains using the bridge, use sovereign accounts on both sides
of the bridge to cover relayer rewards.
```
?

> Bridged Parachains will have a sovereign accounts at bridge hubs. For example, the Statemine (Kusama Parachain) will

Reviewer: ```suggestion
Bridged Parachains will have sovereign accounts at bridge hubs. For example, the Statemine (Kusama Parachain) will
```

have an account at the Polkadot Bridge Hub. The Statemint (Polkadot Parachain) will have an account at the Kusama
Bridge Hub. The sovereign accounts are used as a source of funds when the relayer is calling the
`pallet_bridge_relayers::claim_rewards`.

> Since messages lane is only used by the pair of parachains, there's no collision betweed different bridges. E.g.

Reviewer: ```suggestion
Since messages lane is only used by the pair of parachains, there's no collision between different bridges. E.g.
```

Statemine will only reward relayers that are delivering messages from Statemine. The Statemine sovereign account
is not used to cover rewards of bridging with some other Polkadot Parachain.

### Multiple Relayers and Rewards

> Our goal is to incentivize running honest relayers. But we have no any relayers sets, so at any time anyone may submit

Reviewer: ```suggestion
Our goal is to incentivize running honest relayers. But we have no relayers sets, so at any time anyone may submit
```

> # Polkadot <> Kusama Bridge Overview

Author: To all reviewers: please review this document carefully - I'm describing my own impression of the bridge, which may be different from our actual plans.
